x-pack/plugin/inference/src/yamlRestTest/resources/rest-api-spec/test/inference/30_semantic_text_query.yml

```diff
@@ -0,0 +1,134 @@
+setup:
+  - skip:
+      version: " - 8.12.99"
+      reason: semantic_text introduced in 8.13.0 # TODO change when 8.13.0 is released
+
+  - do:
+      inference.put_model:
+        task_type: sparse_embedding
+        inference_id: sparse-inference-id
+        body: >
+          {
+            "service": "test_service",
+            "service_settings": {
+              "model": "my_model",
+              "api_key": "abc64"
+            },
+            "task_settings": {
+            }
+          }
+
+  - do:
+      inference.put_model:
+        task_type: text_embedding
+        inference_id: dense-inference-id
+        body: >
+          {
+            "service": "text_embedding_test_service",
+            "service_settings": {
+              "model": "my_model",
+              "dimensions": 10,
+              "api_key": "abc64"
+            },
+            "task_settings": {
+            }
+          }
+
+  - do:
+      indices.create:
+        index: test-sparse-index
+        body:
+          mappings:
+            properties:
+              inference_field:
+                type: semantic_text
+                model_id: sparse-inference-id
+              non_inference_field:
+                type: text
+
+  - do:
+      indices.create:
+        index: test-dense-index
+        body:
+          mappings:
+            properties:
+              inference_field:
+                type: semantic_text
+                model_id: dense-inference-id
+              non_inference_field:
+                type: text
+
+---
+"Query using a sparse embedding model":
+  - do:
+      index:
+        index: test-sparse-index
+        id: doc_1
+        body:
+          inference_field: "inference test"
+          non_inference_field: "non inference test"
+        refresh: true
+
+  - do:
+      search:
+        index: test-sparse-index
+        body:
+          query:
+            semantic_query:
+              inference_field:
+                query: "inference test"
+
+  - match: { hits.total.value: 1 }
+  - match: { hits.hits.0._id: "doc_1" }
+  - match: { hits.hits.0._score: 9.4934225 }
```

Comment:
These tests are flaky as written because the score varies slightly (but it's always ~9.49). We should use the `lt` and `gt` checks to check that the score is in the expected range, but those checks inconsistently cast the document score to `Double` or `Float`, making them flaky as well.

IMO we should fix the `lt` & `gt` casting issues in a follow-up PR and disable the score checks here in the meantime.

Reply:
That's weird, as [some tests](https://github.com/elastic/elasticsearch/blob/4b5585e428732b5ac14b6a363bf8f4c6e533e2bb/modules/lang-painless/src/yamlRestTest/resources/rest-api-spec/test/painless/150_dense_vector_l1l2.yml#L91) use lt & gt and are not flaky 🤔

Reply:
Figured this one out: The tests you linked to explicitly force the content type to `application/json`, which forces use of an XContent parser that decodes all floating-point values to `Double`. 

Without this, the content type is randomly chosen by the test and [parsers for some content types](https://github.com/elastic/elasticsearch/blob/392c497551cab6ab947f7995f8bd363101f23fc9/libs/x-content/src/test/java/org/elasticsearch/xcontent/XContentParserTests.java#L70) decode floating-point values to `Float`.